fabric-rendering-v1/src/main/java/net/fabricmc/fabric/impl/client/rendering/ArmorRendererRegistryImpl.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.client.rendering;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.Item;
+import net.minecraft.util.registry.Registry;
+
+import net.fabricmc.fabric.api.client.rendering.v1.ArmorRenderer;
+
+public class ArmorRendererRegistryImpl {
+	private static final HashMap<Item, ArmorRenderer> RENDERERS = new HashMap<>();
+
+	public static void register(ArmorRenderer renderer, ItemConvertible... items) {
+		Objects.requireNonNull(renderer, "renderer is null");
+
+		if (items.length == 0) {
+			throw new IllegalArgumentException("Armor renderer registered for no item");
+		}
+
+		for (ItemConvertible item : items) {
+			Objects.requireNonNull(item, "armor item is null");
```

Comment:
```suggestion
			Objects.requireNonNull(item.asItem(), "armor item is null");
```

Reply:
Ideally both of them, but `item.asItem()` will conveniently NPE if the convertible is null.

Reply:
If an implementation is bad enough that `asItem` returns `null`, I would expect it to be bad enough that it returns different `Item` instances on different `asItem` calls... Should we keep the retrieved item in a local var hence